discovery-provider/src/utils/db_session.py

```diff
@@ -17,35 +12,20 @@ def on_init_db(db):
         set_search_similarity(session)
 
 
-
 def get_db():
     """Connect to the configured database. The connection
     is unique for each request and will be reused if this is called
     again.
     """
-    if "db" not in g:
-        g.db = SessionManager(
-            current_app.config["db"]["url"],
-            ast.literal_eval(current_app.config["db"]["engine_args_literal"]),
-        )
-        on_init_db(g.db)
-
-    return g.db
+    return current_app.db_session_manager
```

Comment:
this cross-dependency is kind of weird
`db_session.py` relies on session manager props being set in `__init__.py` file and `__init__py` file relies on `SessionManager` and `on_init_db()` from this file

Reply:
yeah fair point, i can move session manager out into its own class w/ init db